src/microbe_stage/Microbe.cs

```diff
@@ -2108,6 +2110,82 @@ private Transform GetNewPhysicsRotation(Transform transform)
         return new Transform(new Basis(slerped), transform.origin);
     }
 
+    /// <summary>
+    ///   Adds torque to reach the targeted rotation
+    /// </summary>
+    private void ApplyPhysicsRotation(PhysicsDirectBodyState physicsState)
+    {
+        var transform = physicsState.Transform;
+        var target = transform.LookingAt(LookAtPoint, new Vector3(0, 1, 0));
+        if (transform == target)
+            return;
+
+        var strength = Constants.CELL_BASE_THRUST + Constants.CELL_BASE_THRUST * (HexCount/2);
+        var maxVelocity = 5.0f;
+
+        var distance = target.basis.GetEuler().y - transform.basis.GetEuler().y;
+        distance = (float)(distance * (180/Math.PI));
+        if (distance > 180) {
+            distance = -360 - distance;
+        } else if (distance < -180) {
+            distance = 360 + distance;
+        }
+
+        var force = 0.0f;
+        var velocity = physicsState.AngularVelocity.y;
+
+        if (velocity < -maxVelocity) {
+            force = 1;
+        } else if (velocity > maxVelocity) {
+            force = -1;
+        } else {
+            float velocityTo90 = 0.0f;
+            if (velocity != 0)
+                velocityTo90 = ((velocity)/maxVelocity) * 90;
+            
+            if (velocityTo90 < 0.1f && velocityTo90 > 0.0f || velocityTo90 > -0.1f && velocityTo90 < 0.0f ){
+                if (distance > 2.0f) {
+                    force = 1;
+                }
+                else if (distance < -2.0f) {
+                    force = -1;
+                }
+            } else if (distance >= 90 && velocity < maxVelocity * 0.8f) {
+                force = 1;
+            }
+            else if (distance <= -90 && velocity > -maxVelocity * 0.8f) {
+                force = -1;
+            }
+            else if (velocityTo90 > 0) {
+                if (0 < distance && distance < 90 && velocityTo90/distance > 0.7f) {
+                    force = -(velocityTo90/distance) * 1.5f;
+                } else if (2.0f < distance && distance < 90 && velocityTo90/distance < 0.7f) {
+                    force = (distance/velocityTo90);
+                } else if (-2.0f > distance && distance > -90) {
+                    force = -90/velocityTo90;
+                }              
+            } else if (velocityTo90 < 0) {
+                if (0 > distance && distance > -90 && velocityTo90/distance > 0.7f) {
+                    force = (velocityTo90/distance) * 1.5f;
+                }
+                else if (-2.0f > distance && distance > -90 && velocityTo90/distance < 0.7f) {
+                    force = -(distance/velocityTo90);
+                }
+                else if (2.0f < distance && distance < 90) {
+                    force = -90/velocityTo90;
```

Comment:
I really don't get what motivate this whole bunch of computations. Are these known formulae? Is this a discretization of computations? By the way, why not using more continuous computations?

A few comments would be very welcome to explain things at this point.

Reply:
I just thought about what cases would be possibly and what would be the best way to handle them. After that I experimented with the numbers to make the gameplay as smooth as possible.

Reply:
Thrive is generally trying to be as scientifically accurate with computations as possible, so it would be better to have a proper model of what happens. I think you should have a few lines describing the general behavior of your rotation process model, which is not completely obvious from what I read.

Reply:
Also, for number you played with and whose values you tweaked, these should be defined in `Constants.cs`.

Reply:
Degrees vs radians doesn't really matter as it is just one float multiplication away. Though if you go radians -> degrees -> radians, I think that's bad and you shouldn't add that intermediate conversion instead.
In general I want angles in the game to user quaternions, because euler angles (even if they are radians), have gimbal lock problems. So any serious work should use quaternions. I know that due to microbes being locked to an axis gimbal lock probably can't happen here, but I'd still like to see the `basis` being used in computations *without* stripping away information by converting it to euler angles.